protocol.txt

```diff
@@ -123,3 +127,93 @@ MSG_PACKET_V2
+       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    12 |       9       |       0       |               0               |
+       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
+    Sends a signal to esp to stop a wifi scan. The last scanning segment
+    is cancelled and the esp sends the last update of scan result.
+
+
+MSG_SCAN_AP_CNT
+    This message is only sent by the ESP.
+
+        0               1               2               3
+        0 1 2 3 4 5 6 7 0 1 2 3 4 5 6 7 0 1 2 3 4 5 6 7 0 1 2 3 4 5 6 7
+       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+     0 |                                                               |
+       +                            Intron                             +
+     8 |                                                               |
+       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    12 |       10      |     COUNT     |               0               |
+       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
+    This message is send every time after the scan segment is
+    compeleted.
+
+    COUNT Number of stored scanned result. 
+
+
+MSG_SCAN_AP_GET
+    This message is sent by both printer and ESP.
+
+        0               1               2               3
+        0 1 2 3 4 5 6 7 0 1 2 3 4 5 6 7 0 1 2 3 4 5 6 7 0 1 2 3 4 5 6 7
+       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+     0 |                                                               |
+       +                            Intron                             +
+     8 |                                                               |
+       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    12 |       11      |    AP_INDEX   |           AP_INFO_LEN         |
+       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    16 |                                                               |
+       ~                                                               ~
+       ~                           AP_INFO                             ~
+       ~                                                               ~
+       |                                                               |
+       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
+    This message is send every time after the scan segment is
+    compeleted.
+
+    AP_INDEX    This value is only send to the ESP by the printer and
+                is not set in the response. This values should be valid
+                index into the internal ap storage
+
+    AP_INFO_LEN Length of send ap info data. The value is fixed in
+                every version (but might change based on the version).
+
+    AP_INFO     Binary representation of the stored ap info in ESP.
+                For more info check the ScanData structure in
+                uart_nic.c.
```

Comment:
Only present when sent by the ESP?

You mean ScanResult? (There's no ScanData there).

Btw, if sent on the wire, it should at least be „packed“. Even though there are only 1-byte fields, the compiler is free to add padding and we are not 100% sure it would do the same padding on both sides (because potentially different compilers).